integration-tests/devtools/src/test/java/io/quarkus/devtools/codestarts/quarkus/QuarkusCodestartGenerationTest.java

```diff
@@ -311,6 +311,49 @@ void generateMavenResteasyJava() throws IOException {
         assertThat(projectDir.resolve("src/test/java/org/acme/NativeGreetingResourceIT.java")).exists();
     }
 
+    @Test
+    void generateMavenPicocliJava() throws IOException {
+        final QuarkusCodestartProjectInput input = QuarkusCodestartProjectInput.builder()
+                .addExtension(AppArtifactKey.fromString("io.quarkus:quarkus-picocli"))
+                .addCodestart("picocli")
+                .addData(getTestInputData())
+                .build();
+        final Path projectDir = testDirPath.resolve("maven-picocli-java");
+        getCatalog().createProject(input).generate(projectDir);
+
+        checkMaven(projectDir);
+        checkReadme(projectDir);
+        checkDockerfiles(projectDir, BuildTool.MAVEN);
+        checkConfigProperties(projectDir);
+
+        assertThat(projectDir.resolve("src/main/java/org/acme/picocli/EntryCommand.java")).exists();
+        assertThat(projectDir.resolve("src/main/java/org/acme/picocli/GoodbyeCommand.java")).exists();
+        assertThat(projectDir.resolve("src/main/java/org/acme/picocli/HelloCommand.java")).exists();
+        assertThat(projectDir.resolve("src/main/java/org/acme/picocli/GreetingService.java")).exists();
+    }
+
+    @Test
+    void generateMavenPicocliKotlin() throws IOException {
+        final QuarkusCodestartProjectInput input = QuarkusCodestartProjectInput.builder()
+                .addExtension(AppArtifactKey.fromString("io.quarkus:quarkus-picocli"))
+                .addExtension(AppArtifactKey.fromString("io.quarkus:quarkus-kotlin"))
+                .addCodestart("picocli")
```

Comment:
You don't need to add the codestart, it will be autoselected with the extension, but first you need to add it to the extension metadata in this file:
https://github.com/quarkusio/quarkus/blob/master/extensions/picocli/runtime/src/main/resources/META-INF/quarkus-extension.yaml#L10

```json
---
name: "Picocli"
metadata:
  keywords:
  - "picocli"
  - "command-line"
  guide: "https://quarkus.io/guides/picocli"
  categories:
  - "miscellaneous"
  status: "experimental"
  codestart: picocli

```

Reply:
Ok, this one is missing from guide I think.

Reply:
@jtama-op https://github.com/ia3andy/quarkus/blob/cs-doc/independent-projects/tools/codestarts/README.adoc#writing-a-codestart-with-example-code-for-an-extension

Look at the When done part :)